jni/minitouch/minitouch.c

```diff
@@ -699,6 +699,71 @@ static void io_handler(FILE* input, FILE* output, internal_state_t* state)
   }
 }
 
+static void proxy_handler(FILE* input, FILE* output, int proxy_fd)
+{
+  char read_buffer[80];
+  FILE* proxy_input;
+  FILE* proxy_output;
+  proxy_input = fdopen(proxy_fd, "r");
+  if (proxy_input == NULL)
+  {
+    fprintf(stderr, "%s: fdopen(proxy_fd,'r')\n", strerror(errno));
+    exit(1);
+  }
+  proxy_output = fdopen(dup(proxy_fd), "w");
```

Comment:
Well ... we could but I am not sure it's very important. If dup fails, it will try to fdopen fildes -1 which will return NULL and this result is properly checked. The error message might indeed be a bit inaccurate but still relevant.
I also wrote it that way cause this is already done on [line 969](https://github.com/openstf/minitouch/blame/3af6c07d2338e150e0afcc1c078762b7f5ce45c3/jni/minitouch/minitouch.c#L969).

Reply:
ok, makes sense 👍